### ISSUE TITLE
Add TimeDistill support for model compression

### ISSUE THREAD

**Issue by unpopularrepo** [kwuking/TimeMixer#173]
## What is the problem?

TimeMixer is very good at forecasting, but the model can be big and slow for some applications. The README mentions a new paper called TimeDistill that makes models smaller and faster using knowledge distillation. We should add this feature to TimeMixer.

## Why is this important?

1. Smaller models run faster
2. Smaller models use less memory
3. Smaller models can run on edge devices and phones
4. Customers with limited computing resources can still use TimeMixer

### 1. Basic distillation pipeline

```python
def distill_model(teacher_model, student_config, train_data, epochs=50):
    """Train a smaller student model using knowledge from teacher model"""
    
    # Create smaller student model
    student_model = TimeMixer(student_config)
    
    # Use special distillation loss
    distiller = TimeDistiller(
        teacher_model=teacher_model,
        student_model=student_model,
        alpha=0.5  # Balance between distillation and direct loss
    )
    
    # Train student using teacher's knowledge
    for epoch in range(epochs):
        for batch in train_data:
            loss = distiller.train_step(batch)
            # Update student model
    
    return student_model
```

### 2. Cross-architecture distillation

Support distilling from TimeMixer to different model types:

```python
class TimeDistiller:
    def __init__(self, teacher_model, student_model, alpha=0.5):
        self.teacher = teacher_model
        self.student = student_model
        self.alpha = alpha
        
    def compute_distillation_loss(self, student_outputs, teacher_outputs):
        """Special loss to make student mimic teacher behavior"""
        # Implementation details
        
    def train_step(self, batch):
        # Run teacher model (no gradient)
        with torch.no_grad():
            teacher_outputs = self.teacher(batch)
        
        # Run student model (with gradient)
        student_outputs = self.student(batch)
        
        # Compute mixed loss
        true_labels = batch["labels"]
        direct_loss = F.mse_loss(student_outputs, true_labels)
        distill_loss = self.compute_distillation_loss(student_outputs, teacher_outputs)
        
        # Combined loss
        total_loss = (1 - self.alpha) * direct_loss + self.alpha * distill_loss
        
        return total_loss
```

### 3. Command-line tool

Add a simple command to run distillation from console:

```
python distill.py --teacher-model saved_models/timemixer_large.pth --student-config configs/timemixer_small.json --output-model distilled_model.pth
```

## How to test it?

1. Train a big TimeMixer model on a dataset
2. Distill to a smaller model (like 50% size)
3. Compare accuracy and speed of both models
4. Check if distilled model is almost as good but much faster

## What will users get?

- Much faster inference: up to 5x speed improvement
- Smaller model files: up to 10x reduction
- Almost the same accuracy as big models (maybe 1-2% worse)
- Ability to deploy TimeMixer in resource-constrained environments

**Comment by kevinliu2000** [comment]
Many thanks for your advice. I understand that TimeDistill has not been officially released yet. Once it is formally released, you may contact the authors to obtain the relevant resources.
